vnext/Microsoft.ReactNative/RedBoxHandler.idl

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import "ReactNativeHost.idl";
+
+namespace Microsoft.ReactNative.RedBox {
+
+  enum ErrorType {
+    JavaScriptFatal, // A JS Exception was thrown or otherwise fatal error
+    JavaScriptSoft, // An error coming from JS that isn't fatal, such as console.error
+    Native,
+  };
+
+  [webhosthidden] interface IErrorFrameInfo {
+    String File { get; };
+    String Method { get; };
+    UInt32 Line { get; };
+    UInt32 Column { get; };
+  }
+  
+  [webhosthidden] interface IErrorInfo {
+    String Message { get; };
+    UInt32 Id { get; };
+    IVectorView<IErrorFrameInfo> Callstack { get; };
+  }
+
+  [webhosthidden]
+  interface IRedBoxHandler 
```

Comment:
The previous implementation, now `DefaultRedBoxHandler` has the comment:
```
This class is implemented such that the methods on IRedBoxHandler are thread safe
```

If this is a requirement from the interface should we document this, or is that implicit in winrt?

Reply:
It should probably be documented.  We're pretty light on the documentation as a whole right now.  @kikisaints - any idea the best way to track that?